trade-finance-manager-ui/server/controllers/case/amendments/requestDateValidation.js

```diff
@@ -0,0 +1,147 @@
+const {
+  isAfter, isBefore, set, getUnixTime,
+} = require('date-fns');
+
+// Checks to see if an element is an object or not
+const isObject = (el) => typeof el === 'object' && el !== null && !(el instanceof Array);
+
+/*
+  Maps through validation errors = require( the server and returns i)t
+  so both Summary Error component and field component
+  can display the error messages correctly.
+*/
+const validationErrorHandler = (errs, href = '') => {
+  const errorSummary = [];
+  const fieldErrors = {};
+
+  if (!errs) {
+    return false;
+  }
+
+  const errors = isObject(errs) ? [errs] : errs;
+
+  errors.forEach((el) => {
+    const mappedErrorMessage = el.errMsg;
+
+    errorSummary.push({
+      text: mappedErrorMessage,
+      href: `${href}#${el.errRef}`,
+    });
+    fieldErrors[el.errRef] = {
+      text: mappedErrorMessage,
+    };
+    if (el.subFieldErrorRefs) {
+      el.subFieldErrorRefs.forEach((subFieldRef) => {
+        fieldErrors[subFieldRef] = {
+          text: mappedErrorMessage,
+        };
+      });
+    }
+  });
+
+  return {
+    errorSummary,
+    fieldErrors,
+  };
+};
+
+/**
+ *
+ * @param {Object} body
+ * @param {Object} facility
+ * @returns {Object} containing errors and amendment date
+ * function to validate the amendment request date
+ * checks if in future or before submission date
+ */
+const amendmentRequestDateValidation = async (body, facility) => {
+  const {
+    'amendment-request-date-day': amendmentRequestDateDay,
+    'amendment-request-date-month': amendmentRequestDateMonth,
+    'amendment-request-date-year': amendmentRequestDateYear,
+  } = body;
+
+  const amendmentRequestDateErrors = [];
+
+  const amendmentRequestIsFullyComplete = amendmentRequestDateDay && amendmentRequestDateMonth && amendmentRequestDateYear;
+  const amendmentRequestIsPartiallyComplete = !amendmentRequestIsFullyComplete
+    && (amendmentRequestDateDay || amendmentRequestDateMonth || amendmentRequestDateYear);
```

Comment:
Compound condition with complexity of 3 or more should be a helper function, but in in the interest of time I am happy for it to be left like this and added to tech-debt so we can look into this at a later stage.